### PR TITLE
Rust demand value lane

### LINKED ISSUE #181: Rust docs: reference/demand value lanes

### PR DESCRIPTION
**Pull request by SirCipher**
Blocked by https://github.com/swimos/swimos-docs/pull/209
Closes #181

### CODE REVIEW

**Review by PiggottDev — APPROVED** [review]

File: src/_rust-server/demand-value-lanes.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  - /backend/demand-value-lanes/
+---
+
+This page covers the specifics of Demand Value Lanes and does not cover the more general aspects of Lanes. For more general information about lanes, see the [Lane]({% link _rust-server/lanes.md %}) page.
+
+# Overview
+
+A Demand Value Lane is a stateless analogue of a [Value Lane]({% link _rust-server/value-lane.md %}). Instead of maintaining a persistent state that can be queried, a Demand Value Lane computes a value, when requested, that is sent to all uplinks attached to it. A Demand Value Lane meets the following requirements:
+
+- Values are `cue`-ed into the Demand Value Lane by calling the [Handler Context's]({% link _rust-server/handler-context.md %}) `cue` function.
```
Comment (horned-sphere) [inline src/_rust-server/demand-value-lanes.md:20]: I think explaining what demand lanes do (computing and emitting a value on demand) would be better before introducing "cue" which is slightly odd terminology.

File: src/_rust-server/demand-value-lanes.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  - /tutorials/demand-value-lanes/
+  - /reference/demand-value-lanes.html
+  - /backend/demand-value-lanes/
+---
+
+This page covers the specifics of Demand Value Lanes and does not cover the more general aspects of Lanes. For more general information about lanes, see the [Lane]({% link _rust-server/lanes.md %}) page.
+
+# Overview
+
+A Demand Value Lane is a stateless analogue of a [Value Lane]({% link _rust-server/value-lane.md %}). Instead of maintaining a persistent state that can be queried, a Demand Value Lane computes a value, when requested, that is sent to all uplinks attached to it. A Demand Value Lane meets the following requirements:
```
Comment (horned-sphere) [inline src/_rust-server/demand-value-lanes.md:18]: Another "meets" to get rid of.

Comment (horned-sphere) [reply src/_rust-server/demand-value-lanes.md:18]: I wouldn't say a deman lane is an analogue of a value lane; they're quite different things. They use pretty much the same protocol (but that's irrelevant to the user).

File: src/_rust-server/demand-value-lanes.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+- Values are `cue`-ed into the Demand Value Lane by calling the [Handler Context's]({% link _rust-server/handler-context.md %}) `cue` function.
+- Following a `cue` invocation, the Demand Value Lane's `on_cue` lifecycle event handler will be invoked for a value to be produced.
+- The cued value will be sent to all attached uplinks.
+
+Demand Value Lanes are ideal for publishing statistical events, where it isn’t important that a client receives every incremental update, only that the client eventually receives the latest state, that the state clients receive is real-time (within the latency of the network), and that updates are desired as often as possible. `cue` invocations may happen at scheduled intervals (using timers) or after another event has been triggered by an agent (such as after another lane receives an update and its lifecycle event handler invokes `cue`).
```
Comment (horned-sphere) [inline src/_rust-server/demand-value-lanes.md:24]: I think I'd say it's good for expensive computations that you don't want running on every update. However, I also don't think that demand lanes, as written, are that useful 🤷

File: src/_rust-server/demand-value-lanes.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        context: HandlerContext<ExampleAgent>,
+    ) -> impl HandlerAction<ExampleAgent, Completion = i32> {
+        context.get_value(ExampleAgent::STATE).map(|n| 2 * n)
+    }
+}
+```
+
+# Use cases
+
+Demand Value Lanes are suitable for situations where you aren't interested in the data immediately and can handle the delay between linking to the lane and the `cue` invocation. Common usecases are:
```
Comment (horned-sphere) [inline src/_rust-server/demand-value-lanes.md:72]: I don't really understand this line.

File: src/_rust-server/demand-value-lanes.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+- Propagating metadata. Metadata may be propagated at set intervals using the [Handler Context's]({% link _rust-server/handler-context.md %}) timer API and calculated on-demand.
```
Comment (horned-sphere) [inline src/_rust-server/demand-value-lanes.md:74]: This is more of an internal use ratehr than something a user would do.

**Review by horned-sphere — APPROVED** [review]

**Pull request merged**